pkg/sources/job_progress_hook.go

```diff
@@ -39,6 +39,7 @@ func NewUnitHook(ctx context.Context, opts ...UnitHookOpt) (*UnitHook, <-chan Un
 		logBackPressure: func() {
 			once.Do(func() {
 				ctx.Logger().Info("back pressure detected in unit hook")
+				backPressureOccurrences.Inc()
 			})
 		},
```

Comment:
The log message is in a `once.Do` to not flood the logs if it happens a lot. It would be better to move the metric outside of that:

```go
logBackPressure: func() {
    once.Do(func() {
        ctx.Logger().Info("back pressure detected in unit hook")
    })
    backPressureOccurrences.Inc()
},
```